### ISSUE TITLE
Include wrappers in Lua script

### ISSUE THREAD

**Issue by riccardoch** [erlio/vernemq#868]
### Environment

- VerneMQ Version: 1.5.0
- OS: Ubuntu 18.04.1 LTS

Is it possible to include wrappers in Lua scripts?
I've installed *bcrypt* wrapper (using `luarocks install bcrypt`) and my test script works perfectly.

I tried to include the wrapper in _mongodb.lua_:
`local bcrypt = require("bcrypt")`
but it doesn't work, and I got this error:
`2018-10-02 11:01:52.405 [error] <0.361.0>@vmq_diversity_script_state:init:96 can't load script "/usr/share/vernemq/lua/auth/mongodb.lua" due to {no_module,<<"bcrypt">>,<<"'./bcrypt.lua' './bcrypt/init.lua' '/usr/share/vernemq/lua/bcrypt.lua' ">>}`

**Comment by larshesel** [comment]
Hi, it's not possible to include external lua components unless they're written 100% in lua and are lua 5.2 compatible. We're using the luerl lua interpreter https://luerl.org/.

We do have bcrypt exposed already and you should be able to use it like this: https://github.com/erlio/vernemq/blob/master/apps/vmq_diversity/test/bcrypt_test.lua

Hope this helps

**Comment by riccardoch** [comment]
I knew about your bcrypt implementation, but I need to include other external modules for my purposes.
Anyway thanks for the response!

**Comment by larshesel** [comment]
In general you can `require` stuff. See https://github.com/erlio/vernemq/blob/master/apps/vmq_diversity/priv/auth/mongodb.lua which requires `auth/auth_commons` (`https://github.com/erlio/vernemq/blob/master/apps/vmq_diversity/priv/auth/auth_commons.lua`).

Which functionality is it you'd like to include?

**Comment by riccardoch** [comment]
For my purposes [lua-mongo](https://github.com/neoxic/lua-mongo) library would be perfect, but I can't include it in mongodb.lua.

**Comment by larshesel** [comment]
What about the mongo driver we already expose? Here's our testscript which demonstrates what can be done with it: https://github.com/erlio/vernemq/blob/master/apps/vmq_diversity/test/mongodb_test.lua

What we can expose into lua are erlang libraries, so we're limited to those - the mongo driver we use at the moment is this one: https://github.com/comtihon/mongodb-erlang - does that one support something you need which isn't yet exposed?

**Comment by riccardoch** [comment]
I think I've found the issue with my data model. For every object in my MongoDB I've two date fields ("updatedAt" and "createdAt"); is it possible that _mongodb-erlang_ doesn't support date fields?
If I remove those fields everything works, as I restore them `mongodb.find_one` doesn't retrieve the user object.

**Comment by riccardoch** [comment]
Example of client object in my data model:

```
{
        "username": "riccardo",
        "objectId": "f1VDf7ZWhX",
        "publish_acl": [],
        "subscribe_acl": [],
        "passhash": "*****",
        "className": "vmq_acl_auth",
        "__type": "Object",
        "client_id": "test",
        "updatedAt": "2018-10-02T09:47:03.760Z",
        "createdAt": "2018-10-01T14:52:21.471Z",
        "mountpoint": ""
    }
```

**Comment by larshesel** [comment]
I wonder if that is the same issue as #857 ?

**Comment by riccardoch** [comment]
I think it's the same issue

**Comment by larshesel** [comment]
Are you using ISODate or DATE to insert the dates? If you're inserting plain strings, then it's likely a different issue..

**Comment by riccardoch** [comment]
The date is in ISO 8601 format and it can be saved as:

```
{
 "__type": "Date",
 "iso": "2018-10-02T15:59:11-07:00"
}
```

or as `ISODate("2018-09-27T10:00:06.763Z")`

**Comment by larshesel** [comment]
ok - so this is internally in mongo a datatype of type `Date` so it's the same issue as #857 - I'll close this issue then. Note I pushed a fix in #875 - it would be great if you could test it out and see if it works for you.

**Comment by riccardoch** [comment]
@larshesel I've compiled your vmq_diversity_utils.erl and succesfully tested! Now it works as expected. Thanks for your help 🙏
